File: atcoder/369/B.py
def main():
    n = int(input())
    pos = [-1, -1]
    ans = 0
    for i in range(n):
        a, s = input().split()
        hand = (s == 'R')
        if pos[hand] != -1:
            ans += abs(pos[hand] - int(a))
        pos[hand] = int(a)
    print(ans)
if __name__ == '__main__':
    main()

File: atcoder/369/B.cpp
#include <iostream>
#include <vector>
using namespace std;
int main()
{
    int n;
    cin >> n;
    vector<int> pos{-1, -1};
    int ans = 0;
    for (int i = 0; i < n; i++)
    {
        int a;
        char s;
        cin >> a >> s;
        int hand = (s == 'R');
        if (pos[hand] != -1)
        {
            ans += abs(pos[hand] - a);
        }
        pos[hand] = a;
    }
    cout << ans << endl;
    return 0;
}